File: 프로그래머스/1/12933. 정수 내림차순으로 배치하기/정수 내림차순으로 배치하기.cpp
#include <string>
#include <vector>
#include <bits/stdc++.h>
using namespace std;

vector<long long> v;
long long solution(long long n) {
    long long answer = 0;
    while(n > 0){
        v.push_back(n % 10);
        n /= 10;
    }
    sort(v.begin(), v.end(), greater<long long>());
    for(long long i = 0; i < v.size(); i++){
        if(v[i] == 0)
            continue;
        else
            answer += v[i] * (long long)pow(10, v.size() - 1 -i); 
    }
    return answer;
}

File: 프로그래머스/1/12933. 정수 내림차순으로 배치하기/정수 내림차순으로 배치하기.java
import java.util.*;
class Solution {
    public long solution(long n) {
        long answer = 0;
        List<Long> li = new ArrayList<Long>();
        while(n > 0){
            li.add(n % 10);
            n /= 10;
        }
        Collections.sort(li, Collections.reverseOrder());
        for(int i = 0; i < li.size(); i++){
            if(li.get(i) == 0)
                continue;
            else
                answer += li.get(i) * (long)Math.pow(10, li.size() - 1 -i); 
    }
    return answer;
    }
}